test/integration-serial/serial_deploy_test.rb

```diff
@@ -44,70 +44,6 @@ def test_create_secrets_from_ejson
     refute_logs_match(Base64.strict_encode64(ejson_cloud.catphotoscom_key_value))
   end
 
-  # This can be run in parallel when we switch to --kubeconfig (https://github.com/Shopify/kubernetes-deploy/issues/52)
-  def test_unreachable_context
```

Comment:
I really like the change to make `KubeclientBuilder` a class, and the other two tests removed here were about such granular behaviour that I don't feel strongly about their conversion to the unit level. However, I think we should keep this one and just accept that it will need to stay in the serial suite unless we decide to change the tasks' APIs in the future. We don't have any other tests that simulate this "Can't connect" behaviour.

Reply:
So, this PR is not currently delivering any of the value it was supposed to. It was intended to make these integration tests executable in parallel, but instead it's ending up converting two into unit tests and leaving one running in serial.

Is any of this worth doing at all? What would you actually like to get from this?

Reply:
Thanks for taking a step back and bringing this up. ❤️ You're totally right that we aren't really accomplishing our original goals. I think this is a case where a quick idea we had turned out differently than expected when actually implemented. Looking where this PR ended up, I see the following wins:
- Local testing indicates that converting the two tests to the unit level will save us ~1.7 seconds each (the one that needs to stay happens to be faster, at ~0.5 seconds).
- The refactor to make the `KubeclientBuilder` a class is an improvement in code quality. I can envision further improvements to make the class smarter in the future, now that it can hold state itself.

Although different from the original goals, these improvements seem worth merging to me.

Reply:
Done.